src/content/guides/asset-modules.md

```diff
@@ -0,0 +1,353 @@
+    path: path.resolve(__dirname, 'dist')
+  },
+  experiments: {
+    asset: true
+  },
+  module: {
+    rules: [
+      {
+        test: /\.svg/,
+        type: 'asset/inline',
++       generator: {
++         dataUrl: content => {
++           content = content.toString();
++           return svgToMiniDataURI(content);
++         }
++       }
+      }
+    ]
+  }
+};
+```
+
+This one will use `mini-svg-data-uri` package to encode all `svg` files.
+
+## Source assets
+
+__webpack.config.js__
+
+``` diff
+const path = require('path');
+- const svgToMiniDataURI = require('mini-svg-data-uri');
+
+module.exports = {
+  entry: './src/index.js',
+  output: {
+    filename: 'main.js',
+    path: path.resolve(__dirname, 'dist')
+  },
+  experiments: {
+    asset: true
+  },
+  module: {
+    rules: [
+      {
+-       test: /\.svg/,
+-       type: 'asset/inline',
+-       generator: {
+-         dataUrl: content => {
+-           if (typeof content !== "string") {
+-             content = content.toString();
+-           }
+-
+-           return svgToMiniDataURI(content);
+-         }
+-       }
++       test: /\.txt/,
++       type: 'asset/source',
+      }
+    ]
+  }
+};
+```
+
+__src/example.txt__
+
+``` text
+Hello world
+```
+
+__src/index.js__
+
+``` diff
+- import metroMap from './images/matro.svg';
++ import examleText from './example.txt';
+
+// ...
+
+- block.style.background = `url(${metroMap}); // url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDo...vc3ZnPgo=)
++ block.textContent = examleText; // 'Hello world'
+```
+
+All `txt` files will be injected into the bundles as is. Just like with `raw-loader`
+
+## General asset type
+
+__webpack.config.js__
+
+``` diff
+const path = require('path');
+
+module.exports = {
+  entry: './src/index.js',
+  output: {
+    filename: 'main.js',
+    path: path.resolve(__dirname, 'dist')
+  },
+  experiments: {
+    asset: true
+  },
+  module: {
+    rules: [
+      {
++       test: /\.txt/,
++       type: 'asset',
+      }
+    ]
+  }
+};
+```
+
+This one will automatically chooses between `resource` and `inline` by a condition.
+
+Followed by default condition, for all the files with size less than 8kb will be applied `inline` module type, and `resource` otherwise.
+
+You may customize the condition by the parser options:
+
+__webpack.config.js__
+
+``` diff
+const path = require('path');
+
+module.exports = {
+  entry: './src/index.js',
+  output: {
+    filename: 'main.js',
+    path: path.resolve(__dirname, 'dist')
+  },
+  experiments: {
+    asset: true
+  },
+  module: {
+    rules: [
+      {
+        test: /\.txt/,
+        type: 'asset',
++       parser: {
++         dataUrlCondition: {
++           maxSize: 4 * 1024 // 4kb
++         }
++       }
+      }
+    ]
+  }
+};
+```
+
+Followed by this condition, for all the files with size less than 4kb will be applied `inline` module type, and `resource` otherwise.
+
+Also you can specify a function to decide to inline or not:
```

Comment:
```suggestion
This doesnt belong here.. it should link to a configuration page with appropriate example of how to set dataUrlCondition to a function
```

Reply:
Why? `Configuration` section describes the options, but `Guides` contains the examples of using it.
Am I wrong?

Reply:
Configuration describes the options and provides one or more example of changing it to possible values. Guide section is about step by step demonstration of achieving the topic, described in the guide. What you do here, is exactly going through possible values of a configuration option. This guide piece could link to a configuration page where this option configured as a function is described, including types etc.